src/admin/context.ts

```diff
@@ -75,29 +76,61 @@ export class ContextManager implements IContext {
   }
 
   get authenticatedUser(): AdminAPIUser {
-    const accessGroups = this.config.request.headers.groups.split(',');
+    // If anyone decides to work with/test the subgraph directly,
+    // make sure we cater for undefined headers.
+    const groups = this.config.request.headers.groups as string;
+    const accessGroups = groups ? groups.split(',') : [];
 
     const hasFullAccess = accessGroups.includes(
       MozillaAccessGroup.SCHEDULED_SURFACE_CURATOR_FULL
     );
     const hasReadOnly = accessGroups.includes(MozillaAccessGroup.READONLY);
 
     const user: AdminAPIUser = {
-      name: this.config.request.headers.name,
-      username: this.config.request.headers.username,
+      name: this.config.request.headers.name as string,
+      username: this.config.request.headers.username as string,
       groups: accessGroups,
       hasFullAccess,
       hasReadOnly,
-      // Whether the authenticated user can run queries for a given Scheduled Surface
+      // Whether the authenticated user can run queries for a given scheduled surface
       canRead: (scheduledSurfaceGuid: string): boolean =>
-        hasReadOnly || user.canWrite(scheduledSurfaceGuid),
+        hasReadOnly || user.canWriteToSurface(scheduledSurfaceGuid),
 
-      // Whether the authenticated user can execute mutations for a given Scheduled Surface
-      canWrite: (scheduledSurfaceGuid: string): boolean => {
+      // Whether the authenticated user can execute mutations for corpus entities
+      canWriteToCorpus: (): boolean => {
+        // Full access to everything is an automatic "Yes".
+        if (hasFullAccess) {
+          return true;
+        }
+
+        // As long as the user has access to a specific scheduled surface,
+        // they can create/edit/delete corpus items, too.
+        const scheduledSurfaceAccessGroups = [
+          MozillaAccessGroup.NEW_TAB_CURATOR_ENUS as string, // Oh, TypeScript!
+          MozillaAccessGroup.NEW_TAB_CURATOR_ENGB as string,
+          MozillaAccessGroup.NEW_TAB_CURATOR_DEDE as string,
+          MozillaAccessGroup.NEW_TAB_CURATOR_ENINTL as string,
+          MozillaAccessGroup.POCKET_HITS_CURATOR_ENUS as string,
+          MozillaAccessGroup.POCKET_HITS_CURATOR_DEDE as string,
+        ];
+
+        return (
+          accessGroups.filter((element) => {
```

Comment:
consider set intersection as a simpler alternative

Reply:
uhh, nvm, javascript does not have simple set intersection???????

Reply:
Exactly, Kat!

Reply:
we could maybe use `filter` here? but that's a nit-pick and doesn't block.